test/guard.js

```diff
@@ -425,3 +425,50 @@ tap.test(
 		});
 	},
 );
+
+tap.test("Guard() - do not add metric when a label value is undefined and emit warning", (t) => {
```

Comment:
An `undefined` value should be allowed in my opinion, and not emit a warning.

Reply:
now I'm confused 😄 as sending in `undefined` causes an error in `prom-client`, so we want to emit this rather than having it crash.
that was at least what I was trying to do :)